hermes-2.cxx

```diff
@@ -2172,55 +2337,33 @@ int Hermes::rhs(BoutReal t) {
       check_all(Vi);
       Field3D nevi = mul_all(Ne, Vi);
       check_all(nevi);
-      ddt(Ne) -= Div_parP(nevi);
-    }
-
-    //Skew-symmetric form
-
-    // Field3D gparne = Grad_par(Ne);
-    // Field3D dparve = Div_parP(Ve);
-    // gparne.applyBoundary("neumann");
-    // dparve.applyBoundary("neumann");
-    // mesh->communicate(gparne,dparve);
-    // gparne.applyParallelBoundary(parbc);
-    // dparve.applyParallelBoundary(parbc);
-
-    // check_all(gparne);
-    // check_all(dparve);
-
-    // ddt(Ne) -= 0.5 * (Div_par(neve) + mul_all(Ve,gparne) + mul_all(Ne,
-    // dparve));
-    // b -= 0.5 * (Div_par(neve) + mul_all(Ve,gparne) + mul_all(Ne,
-    // dparve));
-    // //ddt(Ne) -= 0.5 * (Div_par(neve) + Ve * Grad_par(Ne) + Ne *
-    // Div_par(Ve)); a = -0.5 * (Div_parP(neve) + Ve * Grad_par(Ne) + Ne *
-    // Div_par(Ve));
-    // auto* coords = mesh->getCoordinates();
-    // for(auto &i :Ne.getRegion("RGN_NOBNDRY")) {
-    //   ddt(Ne)[i] -= (Ne.yup()[i.yp()] * Ve.yup()[i.yp()] /
-    //   coords->Bxy[i.yp()]
-    //               - Ne.ydown()[i.ym()] * Ve.ydown()[i.ym()] /
-    //               coords->Bxy[i.ym()])
-    //     * coords->Bxy[i] / (coords->dy[i] * sqrt(coords->g_22[i]));
-    // }
+      auto tmp = -Div_parP(nevi);
+      if(TE_Ne){
+	TE_Ne_parflow = tmp;
+      }
+      
+      ddt(Ne) += tmp;
+    }
+
   }
 
   if (j_diamag) {
     // Diamagnetic drift, formulated as a magnetic drift
     // i.e Grad-B + curvature drift
     if (!evolve_ni) {
       mesh->communicate(Pe);
-      ddt(Ne) -= fci_curvature(Pe);
+      ddt(Ne) -= fci_curvature(Pe,use_bracket);
     } else {
       mesh->communicate(Pi);
```

Comment:
```suggestion
```
Communicate should probably not be needed. If it is, you likely also need parallel BCs ...